src/test/shell/integration/runfiles_test.sh

```diff
@@ -133,15 +184,102 @@ EOF
 
   # that accounts for everything
   cd ../..
-  assert_equals  9 $(find ${WORKSPACE_NAME} -type l | wc -l)
-  assert_equals  4 $(find ${WORKSPACE_NAME} -type f | wc -l)
-  assert_equals  9 $(find ${WORKSPACE_NAME} -type d | wc -l)
-  assert_equals 22 $(find ${WORKSPACE_NAME} | wc -l)
-  assert_equals 13 $(wc -l < MANIFEST)
+  if "$is_windows"; then
+    assert_equals 10 $(find ${WORKSPACE_NAME} -type l | wc -l)
+    assert_equals  4 $(find ${WORKSPACE_NAME} -type f | wc -l)
+    assert_equals  9 $(find ${WORKSPACE_NAME} -type d | wc -l)
+    assert_equals 23 $(find ${WORKSPACE_NAME} | wc -l)
+    assert_equals 14 $(wc -l < MANIFEST)
+  else
+    assert_equals  9 $(find ${WORKSPACE_NAME} -type l | wc -l)
+    assert_equals  4 $(find ${WORKSPACE_NAME} -type f | wc -l)
+    assert_equals  9 $(find ${WORKSPACE_NAME} -type d | wc -l)
+    assert_equals 22 $(find ${WORKSPACE_NAME} | wc -l)
+    assert_equals 13 $(wc -l < MANIFEST)
+  fi
+
+  for i in $(find ${WORKSPACE_NAME} \! -type d); do
+    if readlink "$i" > /dev/null; then
+      if "$is_windows"; then
+        echo "$i $(cygpath -m $(readlink "$i"))" >> ${TEST_TMPDIR}/MANIFEST2
+      else
+        echo "$i $(readlink "$i")" >> ${TEST_TMPDIR}/MANIFEST2
+      fi
+    else
+      echo "$i " >> ${TEST_TMPDIR}/MANIFEST2
+    fi
+  done
+  sort MANIFEST > ${TEST_TMPDIR}/MANIFEST_sorted
+  sort ${TEST_TMPDIR}/MANIFEST2 > ${TEST_TMPDIR}/MANIFEST2_sorted
+  diff -u ${TEST_TMPDIR}/MANIFEST_sorted ${TEST_TMPDIR}/MANIFEST2_sorted
+
+  # Rebuild the same target with a new dependency.
+  cd "$workspace_root"
+cat > $pkg/BUILD << EOF
+sh_binary(name = "foo",
+          srcs = [ "x/y/z.sh" ],
+          data = [ "e/f" ])
+EOF
+  bazel build $pkg:foo $EXTRA_BUILD_FLAGS >&$TEST_log || fail "build failed"
+
+  cd ${PRODUCT_NAME}-bin/$pkg/foo${EXT}.runfiles
+
+  # workaround until we use assert/fail macros in the tests below
+  touch $TEST_TMPDIR/__fail
+
+  # output manifest exists and is non-empty
+  test    -f MANIFEST
+  test    -s MANIFEST
+
+  cd ${WORKSPACE_NAME}
+
+  # these are real directories
+  test \! -L $pkg
+  test    -d $pkg
+
+  # these directory should not exist anymore
+  test \! -e a
+  test \! -e c
+
+  cd $pkg
+  test \! -L e
+  test    -d e
+  test \! -L x
+  test    -d x
+  test \! -L x/y
+  test    -d x/y
+
+  # these are symlinks to the source tree
+  test    -L foo
+  test    -L x/y/z.sh
+  test    -L e/f
+  test    -d e/f
+
+  # that accounts for everything
+  cd ../..
+  if "$is_windows"; then
+    assert_equals  4 $(find ${WORKSPACE_NAME} -type l | wc -l)
```

Comment:
why is there one more symlink on Windows?

Reply:
Answered in above comment.

Reply:
Thanks, please add it as a comment here too.

Reply:
Done